allennlp/commands/predict.py

```diff
@@ -212,10 +221,25 @@ def _get_json_data(self) -> Iterator[JsonDict]:
                     yield self._predictor.load_line(line)
         else:
             input_file = cached_path(self._input_file)
-            with open(input_file, "r") as file_input:
-                for line in file_input:
-                    if not line.isspace():
-                        yield self._predictor.load_line(line)
+            if self.compression_type is None:
+                try:
+                    with open_compressed(input_file) as file_input:
+                        for line in file_input:
+                            if not line.isspace():
+                                yield self._predictor.load_line(line)
+                except OSError:
+                    print(
+                        "Automatic detection failed, please specify the compression type argument."
+                    )
+
+            else:
+                try:
+                    with open_compressed(input_file, compression_type=self.compression_type) as file_input:
+                        for line in file_input:
+                            if not line.isspace():
+                                yield self._predictor.load_line(line)
+                except OSError:
+                    print("please specify the correct compression type argument.")
```

Comment:
Why `OSError`? I don't think you have to catch this exception at all. If it fails, it fails. The only thing we might want to do is make sure that `open_compressed()` throws exceptions that are understandable.